### PR TITLE
Photon: Find atts for image sizes with "0" width or height

### PR DESCRIPTION
**Pull request by dereksmart**
In cases like TwentySixteen, where their thumbnail size is set to have a 0 (dynamic) height, photon will still need to find the atts so we can give it to srcset and resize appropriately.

cc @kraftbj

### CODE REVIEW

**Comment by kraftbj** [comment]
I'd want to explore this a bit. Do we need the full size image or would adding the conditional to the elseif to resize based on that? I'm not sure the behavior if 0 is passed, but that is probably closer to the intent than a full size height/width. I'll write up unit tests tonight/tomorrow morning and we can reverse-engineer a passing test again.

**Comment by kraftbj** [comment]
Added a first pass at unit tests for `(700,0)` size, both hard/soft crop for before/after upload.

File: class.photon.php [hunk 1]
```diff
@@ -457,7 +457,8 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 				$image_meta = image_get_intermediate_size( $attachment_id, $size );
 
 				// 'full' is a special case: We need consistent data regardless of the requested size.
-				if ( 'full' == $size ) {
+				// We will also need a specific size if thumbnails are set to have dynamic (0 value) height/width. 
+				if ( 'full' == $size || 0 == $image_meta['height'] || 0 == $image_meta['width'] ) {
```
Comment (kraftbj) [inline class.photon.php]: I think these need to be strict `===`. If `! $image_meta`, then `0 == $image_meta['height']` would evaluate true.

File: class.photon.php [hunk 2]
```diff
@@ -464,17 +464,16 @@ public function filter_image_downsize( $image, $attachment_id, $size ) {
 					// for an image that was uploaded before the custom image was added to the theme.  Try to determine the size manually.
 					$image_meta = wp_get_attachment_metadata( $attachment_id );
 					$image_resized = image_resize_dimensions( $image_meta['width'], $image_meta['height'], $image_args['width'], $image_args['height'], $image_args['crop'] );
-					$image_meta['width'] = $image_resized[6];
-					$image_meta['height'] = $image_resized[7];
+					if ( $image_resized ) { // This could be false when the requested image size is larger than the full-size image.
```
Comment (dereksmart) [inline class.photon.php]: I like this much better than checking for w & h == 0

File: tests/php/test_class.jetpack_photon.php [hunk 3]
```diff
@@ -412,6 +446,27 @@ public function test_photon_return_jetpack_hard_undefined_size_dimensions() {
 	 * @covers Jetpack_Photon::filter_image_downsize
 	 * @since 3.8.2
 	 */
+	public function test_photon_return_jetpack_hard_undefined_zero_size_dimensions() {
+		global $content_width;
+		$content_width = 0;
+
+		$test_image = $this->_get_image();
+
+		// Using a custom size, declared before the file was uploaded (thus exists per WP), hard crop defined 700 height, any width.
```
Comment (dereksmart) [inline tests/php/test_class.jetpack_photon.php]: > hard crop defined 700 height, any width.

Do you mean the opposite?  (700w any height)

**Comment by dereksmart** [comment]
couple typos in test comments, but other than that this looks good.  Let's push some squash on it and merge.

**Comment by dereksmart** [comment]
:+1: for squash.  :+1: for changeset. :+1: for testing

**Pull request merged**